packages/approval-controller/src/ApprovalController.ts

```diff
@@ -240,13 +240,13 @@ export class ApprovalController extends BaseControllerV2<
    * @returns The approval promise.
    */
   async addAndShowApprovalRequest(opts: AddApprovalOptions): Promise<unknown> {
-    const promise = this._add(
+    const promise = this.#add(
       opts.origin,
       opts.type,
       opts.id,
       opts.requestData,
     );
-    this._showApprovalRequest();
+    await this.#showApprovalRequest();
```

Comment:
This potentially changes the behaviour of this function. We may want to do this instead:

```suggestion
    this.#showApprovalRequest().catch(console.error);
```

Reply:
I think this makes sense.

Reply:
Aside: Original looks like it could be a mistake? :thinking: 

I'd expect the two promises (`this.#showApprovalRequest` and `this.#add`) to be either serially resolved (`await` or `then`), or returned together via a `Promise.all`?

Reply:
It may be, but that's probably something we can decide later.